### ISSUE TITLE
Edit Widgets: Settings toggle shows wrong icon when 'Show button text labels' is on

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#57646]
### Description

In the Edit Widgets page, the Settings panel toggle button shows a wrong icon when 'Show button text labels' is on.

It appears the Edit Widgets page is a bit behind and maybe a little overlooked. 
To start with, it should fullly support 'Show button text labels', like the two other editors.
Anyways, when 'Show button text labels' is enabled in the _post_ editor, the _widgets_ editor shows a wrong icon instead of text.

See:

https://github.com/WordPress/gutenberg/blame/302c1487c0298a75b4a0094e77ec87256f15ccdf/packages/interface/src/components/complementary-area/index.js#L190

```
icon={ showIconLabels ? check : icon }
```

I'm not sure why the Complementary area toggle is supposed to show the 'check' icon when `showIconLabels` is true. There's no apparent reason for that, unless I'm missing something. Cc @tellthemachines 
It seems to be there since https://github.com/WordPress/gutenberg/pull/24304

I'm not even shure the other editors use the `ComplementaryArea` and `ComplementaryAreaToggle` this way.

### Step-by-step reproduction instructions

- Set Twenty Twenty-One as active theme.
- Go to Appearance > Widgets 
- Observe the button to toggle the Settings panel shows the correct icon `drawerRight`.
- Hover or focus the button and observe it shows a tooltip with text 'Settings'.
- Screenshot:

![Screenshot 2024-01-08 at 15 29 19](https://github.com/WordPress/gutenberg/assets/1682452/d9ff4d0c-2524-4b76-a573-014cc972cad0)

- Go to the post editor, and enable Options > Preferences > Accessibility > Show button text labels.
- Go to Appearance > Widgets 
- Observe the button to toggle the Settings panel now shows a wrong 'check' icon.
- Hover or focus the button and observe there is no tooltip.
- Screenshot:

![Screenshot 2024-01-08 at 15 25 30](https://github.com/WordPress/gutenberg/assets/1682452/08e8dcda-0ccd-4b5f-95b3-e5d0e16a9330)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by tellthemachines** [comment]
Thanks for opening this! It is a bug, due to support for button text labels never having been implemented in the widgets editor. Now that it's a cross-editor preference it should be made to work correctly in this editor too.

**Comment by afercia** [comment]
@karthick-murugan thanks for your PR at https://github.com/WordPress/gutenberg/pull/68116
I'm not sure this hsould be fixed by changing how the `ComplementaryArea` toggle icon and text work, at least not for now.

As @tellthemachines mentioned, the broader issue is that support for the `showIconLabels` preference hasn't been implemented in the widgets editor at all. 

The current mechanism to hide the SVG icons and display text is CSS-based, see the implementation in the `editor` pagkage here:

https://github.com/WordPress/gutenberg/blob/aecb645fc3a5c7eba23e0266c7012bcf81fa6284/packages/editor/src/components/header/style.scss#L121-L134

I'm not saying that's ideal. I would prefer a more solid, programamtic, approach like the one you tried on https://github.com/WordPress/gutenberg/pull/68116 but for now I'd think it's best to keep the implementation consistent and use the current CSS approach. A better, more solid, implementation should be considered in the context of the following dedicated issue: https://github.com/WordPress/gutenberg/issues/61763

It's also worth noting that the implementation in the post editor and in the site editor is inconsistent.

In the post editor, the `show-icon-labels` CSS class [is added to the body element](https://github.com/WordPress/gutenberg/blob/900862e0ae020eeb94be098333703b8f26db3be0/packages/edit-post/src/components/layout/index.js#L468-L473) with the reason that it needs to be used also for contant inside modal dialogs. Previously, it was set on the `edit-post-layout` element. That changed in https://github.com/WordPress/gutenberg/pull/53835.

Instead, in the site editor it's still added to [the edit-site-layout element](https://github.com/WordPress/gutenberg/blob/900862e0ae020eeb94be098333703b8f26db3be0/packages/edit-site/src/components/layout/index.js#L101-L108).

I'll submit a PR shortly to start adding support for showIconLabels in the Widgets editor.

**Comment by karthick-murugan** [comment]
@afercia - Thanks for the detailed explanation. In that case, can I close my PR?

**Comment by afercia** [comment]
> [@afercia](https://github.com/afercia) - Thanks for the detailed explanation. In that case, can I close my PR?

@karthick-murugan yes thanks. Feedback and thoughts on the alternative PR https://github.com/WordPress/gutenberg/pull/68531 welcome.
